shared/src/panel/views/HierarchicalLocationsView.tsx

```diff
@@ -78,15 +77,11 @@ export class HierarchicalLocationsView extends React.PureComponent<HierarchicalL
                             switchMap(locations =>
                                 locations.pipe(
                                     catchError((error): [ErrorLike] => [asError(error)]),
-                                    map(result => ({
-                                        locationsOrError: isErrorLike(result)
-                                            ? result
-                                            : { results: result || [], loading: true },
-                                    })),
-                                    startWith<Pick<State, 'locationsOrError'>>({
-                                        locationsOrError: { loading: true },
-                                    }),
-                                    tap(({ locationsOrError }) => {
+                                    map(result =>
+                                        isErrorLike(result) ? result : { results: result || [], loading: true }
+                                    ),
+                                    startWith<State['locationsOrError']>({ loading: true }),
```

Comment:
Also reset `results`?

```suggestion
                                    startWith<State['locationsOrError']>({ loading: true, results: undefined }),
```